### PR TITLE
Implementation of interactive todo list features

### PR DESCRIPTION
**Pull request by serengia**
**Request to merge interactive-list branch to main**
---
The interactive branch contains the following features.
- Checking off items
- Clear checked-off list functionality
- Update UI feature

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -74,3 +76,20 @@ todoContainer.addEventListener("click", (e) => {
...
+
+let indexesToRemove = [];
+
+todoContainer.addEventListener("click", (e) => {
+  const activeEl = e.target.closest(".checkbox");
+  if (!activeEl) return;
+  const parentEl = activeEl.closest(".todo");
+
+  indexesToRemove = checkoffItem(parentEl);
+});
```
Comment (ShahierNashaat) [inline src/index.js:88]: - [ ] Kindly notice that you should update the `localstorage` data when the status change.

File: src/interactiveList.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// eslint-disable-next-line import/prefer-default-export
```
Comment (ShahierNashaat) [inline src/interactiveList.js]: - [ ] Kindly fix the linters instead of disabling them.

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @Serengia, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (serengia) [reply src/index.js:88]: @ShahierNashaat  Thank you for the feedback. Please notice, based on the structure of my code, I am effectively saving on the local storage on the line below. Refer to line no. 92 in the code snippet below:
https://github.com/Serengia/to-do-app-with-webpack/blob/78c851259300d7e56f7590d1da801dcaea867eb6/src/index.js#L79-L95

File: src/interactiveList.js (hunk 2 above)
Comment (serengia) [reply src/interactiveList.js]: I have implemented this.

**Review by ichala — APPROVED** [review]
Hi @Serengia  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

-Requirements Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  @ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (ShahierNashaat) [reply src/index.js:88]: Hi, @Serengia What I mean is the status of the task from `completed: false` to `completed: true` should be updated in the local storage, as when I marked the task as completed and refreshed the page the task returned to not completed status.

**Pull request merged**